ipa-server-install fails in FIPS mode

The command ipa-server-install is failing in FIPS mode on Fedora 43 with the repository updates-testing enabled.

The command is failing when calling pkispawn:
```
FINE: TomcatJSS: initialization
FINE: TomcatJSS: certdbDir: /var/lib/pki/pki-tomcat/conf/alias
FINE: TomcatJSS: passwordClass: org.dogtagpki.jss.tomcat.PlainPasswordFile
FINE: TomcatJSS: passwordFile: /var/lib/pki/pki-tomcat/conf/password.conf
FINE: TomcatJSS: serverCertNickFile: /var/lib/pki/pki-tomcat/conf/serverCertNick.conf
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::loadLibrary has been called by org.mozilla.jss.CryptoManager in an unnamed module (file:/usr/share/java/jss/jss.jar)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

java.security.ProviderException: Initialization failed
	at jdk.crypto.cryptoki/sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:388)
	at jdk.crypto.cryptoki/sun.security.pkcs11.SunPKCS11.configure(SunPKCS11.java:125)
	at java.base/sun.security.jca.ProviderConfig.doLoadProvider(ProviderConfig.java:237)
	at java.base/sun.security.jca.ProviderConfig.getProvider(ProviderConfig.java:210)
	at java.base/sun.security.jca.ProviderList.loadAll(ProviderList.java:311)
	at java.base/sun.security.jca.ProviderList.removeInvalid(ProviderList.java:328)
	at java.base/sun.security.jca.Providers.getFullProviderList(Providers.java:186)
	at java.base/java.security.Security.insertProviderAt(Security.java:491)
	at org.mozilla.jss.CryptoManager.initialize(CryptoManager.java:651)
	at org.dogtagpki.jss.tomcat.TomcatJSS.init(TomcatJSS.java:446)
	at org.dogtagpki.server.cli.SubsystemCLI.initializeTomcatJSS(SubsystemCLI.java:43)
	at org.dogtagpki.server.cli.SubsystemDBRemoveCLI.execute(SubsystemDBRemoveCLI.java:62)
	at org.dogtagpki.cli.CommandCLI.execute(CommandCLI.java:58)
	at org.dogtagpki.cli.CLI.execute(CLI.java:353)
	at org.dogtagpki.cli.CLI.execute(CLI.java:353)
	at org.dogtagpki.cli.CLI.execute(CLI.java:353)
	at org.dogtagpki.server.cli.PKIServerCLI.execute(PKIServerCLI.java:93)
	at org.dogtagpki.server.cli.PKIServerCLI.main(PKIServerCLI.java:123)
Caused by: sun.security.pkcs11.wrapper.PKCS11Exception: CKR_GENERAL_ERROR
	at jdk.crypto.cryptoki/sun.security.pkcs11.wrapper.PKCS11.C_Initialize(Native Method)
	at jdk.crypto.cryptoki/sun.security.pkcs11.wrapper.PKCS11$SynchronizedPKCS11.C_Initialize(PKCS11.java:1710)
	at jdk.crypto.cryptoki/sun.security.pkcs11.wrapper.PKCS11.getInstance(PKCS11.java:179)
	at jdk.crypto.cryptoki/sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:342)
	... 17 more
ERROR: CalledProcessError: Command '['/usr/sbin/runuser', '-u', 'pkiuser', '--', '/usr/lib/jvm/jre-25-openjdk/bin/java', '-classpath', '/usr/share/tomcat/bin/tomcat-juli.jar:/usr/share/java/tomcat-servlet-api.jar:/usr/share/pki/ca/webapps/ca/WEB-INF/lib/*:/var/lib/pki/pki-tomcat/common/lib/*:/usr/share/pki/lib/*', '-Djavax.sql.DataSource.Factory=org.apache.commons.dbcp.BasicDataSourceFactory', '-Dcatalina.base=/var/lib/pki/pki-tomcat', '-Dcatalina.home=/usr/share/tomcat', '-Djava.endorsed.dirs=', '-Djava.io.tmpdir=/var/lib/pki/pki-tomcat/temp', '-Djava.util.logging.config.file=/var/lib/pki/pki-tomcat/conf/logging.properties', '-Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager', '-Dcom.redhat.fips=false', 'org.dogtagpki.server.cli.PKIServerCLI', 'ca-db-remove', '--force', '--debug']' returned non-zero exit status 255.
  File "/usr/lib/python3.14/site-packages/pki/server/pkispawn.py", line 594, in main
    deployer.spawn()
    ~~~~~~~~~~~~~~^^
  File "/usr/lib/python3.14/site-packages/pki/server/deployment/__init__.py", line 5802, in spawn
    scriptlet.spawn(self)
    ~~~~~~~~~~~~~~~^^^^^^
  File "/usr/lib/python3.14/site-packages/pki/server/deployment/scriptlets/configuration.py", line 89, in spawn
    deployer.setup_database(subsystem, master_config)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/pki/server/deployment/__init__.py", line 1537, in setup_database
    subsystem.remove_database(force=True)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/pki/server/subsystem.py", line 1283, in remove_database
    self.run(cmd, as_current_user=as_current_user)
    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/pki/server/subsystem.py", line 2268, in run
    return subprocess.run(
           ~~~~~~~~~~~~~~^
        cmd,
        ^^^^
    ...<2 lines>...
        stderr=stderr,
        ^^^^^^^^^^^^^^
        check=True)
        ^^^^^^^^^^^
  File "/usr/lib64/python3.14/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)

```

A recent change that could explain the issue: java-25-openjdk-crypto-adapter-25.0.1.0.8-3.fc43.x86_64

--- a/base/common/python/pki/cli/main.py
+++ b/base/common/python/pki/cli/main.py
@@ -237,9 +237,9 @@ class PKICLI(pki.cli.CLI):
 
         java_fips_cmd = os.getenv('JAVA_FIPS_ENABLED')
         if java_fips_cmd:
-            cmd.extend([
-                java_fips_cmd
-            ])
+            cmd.extend(
+                java_fips_cmd.split(' ')
+            )
 
         # Deal with warnings that alter CI test results.
         cmd.extend([

--- a/base/common/share/etc/pki.conf
+++ b/base/common/share/etc/pki.conf
@@ -16,7 +16,7 @@ export JAVA_HOME
 # For the moment we want this to be false even if we really are
 # in fips mode, because we want the jss prover instead of the sun
 # fips provider to be selected.
-JAVA_FIPS_ENABLED="-Dcom.redhat.fips=false" # Disable FIPS mode
+JAVA_FIPS_ENABLED="-Dcom.redhat.fips=false -Dredhat.crypto-policies=false" # Disable FIPS mode
 
 export JAVA_FIPS_ENABLED
 

--- a/base/server/python/pki/server/cli/__init__.py
+++ b/base/server/python/pki/server/cli/__init__.py
@@ -254,7 +254,7 @@ class PKIServerCLI(pki.cli.CLI):
 
         java_fips_cmd = os.getenv('JAVA_FIPS_ENABLED')
         if java_fips_cmd:
-            cmd.append(java_fips_cmd)
+            cmd.extend(java_fips_cmd.split(' '))
 
         # suppress JNI warnings
         cmd.append('--enable-native-access=ALL-UNNAMED')

--- a/base/server/share/conf/tomcat.conf
+++ b/base/server/share/conf/tomcat.conf
@@ -27,7 +27,7 @@ CATALINA_TMPDIR="[pki_instance_path]/temp"
 #   -Djava.library.path=/usr/lib
 # - parameters to run a java debugger (e. g. - 'eclipse')
 #   -Xdebug -Xrunjdwp:transport=dt_socket,address=8000,server=y,suspend=n -Djava.awt.headless=true -Xmx128M
-JAVA_OPTS="-Dcom.redhat.fips=false"
+JAVA_OPTS="-Dcom.redhat.fips=false -Dredhat.crypto-policies=false"
 
 # What user should run tomcat
 TOMCAT_USER="[pki_user]"

--- a/base/server/upgrade/11.9.0/04-UpdateJDKFIPS.py
+++ b/base/server/upgrade/11.9.0/04-UpdateJDKFIPS.py
@@ -1,0 +1,43 @@
+# Authors:
+#     Marco Fargetta <mfargett@redhat.com>
+#
+# Copyright Red Hat, Inc.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import logging
+import os
+import pki
+
+
+class UpdateJDKFIPS(pki.server.upgrade.PKIServerUpgradeScriptlet):
+    msg = "# Adding -Dredhat.crypto-policies=false for JDK>=25 as "\
+          "SunJSSE+SunPKCS11 conflicts with JSS\n"
+
+    def __init__(self):
+        super(UpdateJDKFIPS, self).__init__()
+        self.message = 'Set -Dredhat.crypto-policies=false in Tomcat configuration'
+
+    def upgrade_instance(self, instance):
+        self.fix_tomcat_config(instance.tomcat_conf)
+        self.fix_tomcat_config('/etc/sysconfig/%s' % instance.name)
+
+    def fix_tomcat_config(self, filename):
+        if not os.path.exists(filename):
+            logging.debug("Unknown file: %s", filename)
+            return
+
+        with open(filename, 'r', encoding='utf-8') as in_fp:
+            lines = in_fp.readlines()
+
+        with open(filename, 'w', encoding='utf-8') as out_fp:
+            for line in lines:
+                if 'JAVA_OPTS="' in line and 'redhat.crypto-policies' not in line:
+                    out_fp.write(self.msg)
+                    line = line.replace(
+                        'JAVA_OPTS="',
+                        'JAVA_OPTS="-Dredhat.crypto-policies=false ',
+                        1
+                    )
+
+                out_fp.write(line)

--- a/base/tools/templates/pki_java_command_wrapper.in
+++ b/base/tools/templates/pki_java_command_wrapper.in
@@ -53,6 +53,7 @@ JAVA_OPTIONS=""
 ${JAVA} ${JAVA_OPTIONS} \
   -cp "${PKI_LIB}/*" \
   -Dcom.redhat.fips=false \
+  -Dredhat.crypto-policies=false \
   -Djava.util.logging.config.file=${PKI_LOGGING_CONFIG} \
   com.netscape.cmstools.${COMMAND} "$@"
 

